bodo/hiframes/boxing.py

```diff
@@ -936,6 +936,9 @@ def _infer_series_arr_type(S: pd.Series, array_metadata=None):
             arr_type = types.Array(arr_type.dtype, 1, "C")
 
         return arr_type
+    except pa.lib.ArrowMemoryError:  # pragma: no cover
+        # OOM
+        raise
```

Comment:
I noticed this when scattering Series that are too large to fit into memory `S.array` can fail. Before the error message would be:
```
bodo.utils.typing.BodoError: data type string for column A not supported yet
``` Now the error looks like:
```
pa.lib.ArrowMemoryError: Could not find an empty frame of required size (68719476736)! 
```